### PR TITLE
TrtWorld add extractor

### PR DESCRIPTION
**Pull request by ufukk**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Extractor for trtworld.com
Resolves: #8455

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by seproDev** [comment]
Please fill out the template checklist or else the PR cannot be accepted

File: yt_dlp/extractor/trtworld.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                "News & Politics"
+            ],
+            "channel": "TRT World",
+            "channel_follower_count": int,
+            "channel_is_verified": True,
+            "uploader": "TRT World",
+            "uploader_id": "@trtworld",
+            "uploader_url": "https://www.youtube.com/@trtworld",
+            "upload_date": "20231202",
+            "availability": "public",
```
Comment (seproDev) [inline yt_dlp/extractor/trtworld.py:68]: Should be single quotes
```suggestion
            'thumbnail': 'https://i.ytimg.com/vi/zEns2dWl00w/maxresdefault.jpg',
            'description': 'md5:3ad9d7c5234d752a4ead4340c79c6b8d',
            'channel_id': 'UC7fWeaHhqgM4Ry-RMpM2YYw',
            'channel_url': 'https://www.youtube.com/channel/UC7fWeaHhqgM4Ry-RMpM2YYw',
            'duration': 210,
            'view_count': int,
            'age_limit': 0,
            'webpage_url': 'https://www.youtube.com/watch?v=zEns2dWl00w',
            'categories': ['News & Politics'],
            'channel': 'TRT World',
            'channel_follower_count': int,
            'channel_is_verified': True,
            'uploader': 'TRT World',
            'uploader_id': '@trtworld',
            'uploader_url': 'https://www.youtube.com/@trtworld',
            'upload_date': '20231202',
            'availability': 'public',
```

File: yt_dlp/extractor/trtworld.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            'comment_count': int,
+            'playable_in_embed': True,
+            'tags': [],
+            'live_status': 'not_live',
+            'like_count': None
+
```
Comment (seproDev) [inline yt_dlp/extractor/trtworld.py:74]: Like count is int
```suggestion
            'like_count': int,
```

File: yt_dlp/extractor/trtworld.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            if youtube_id := traverse_obj(nuxtjs_data, ('platforms', 'youtube', 'metadata', 'youtubeId')):
+                return self.url_result(youtube_id, 'Youtube')
+            raise ExtractorError('No video found', expected=True)
+
+        return {
+            'id': display_id,
+            'formats': formats,
+            **traverse_obj(nuxtjs_data, ('platforms', ('website', 'ott'), {
+                'title': ('fields', 'title', 'text'),
+                'description': ('fields', 'description', 'text'),
```
Comment (Grub4K) [inline yt_dlp/extractor/trtworld.py:97]: I would probably add a `{str}` just to be safe
```suggestion
                'title': ('fields', 'title', 'text', {str}),
                'description': ('fields', 'description', 'text', {str}),
```

File: yt_dlp/extractor/trtworld.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            'tags': [],
+            'live_status': 'not_live',
+            'like_count': int,
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        nuxtjs_data = self._search_nuxt_data(webpage, display_id)['videoData']['content']
```
Comment (Grub4K) [inline yt_dlp/extractor/trtworld.py:75]: Maybe this should also include `'platforms'` as we only ever access that later on?
```suggestion
        nuxtjs_data = self._search_nuxt_data(webpage, display_id)['videoData']['content']['platforms']
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**